packages/fxa-settings/src/lib/model-data/data-stores/generic-data.ts

```diff
@@ -26,3 +26,14 @@ export class GenericData extends ModelDataStore {
     this.state[key] = value;
   }
 }
+
+export function queryStringToGenericData(query: string) {
+  const urlParams = new URLSearchParams(query);
```

Comment:
If the query contains any `+` signs that are not encoded, they will be transformed to spaces by `URLSearchParams`

Reply:
Agree based on discussion we had after I left the comment 👍